### PR TITLE
8263807: Button types of a DialogPane are set twice, returns a wrong button

### PR DESCRIPTION
**Pull request by Maran23**
When DialogPane#getButtonTypes().setAll() is called twice with the same argument(s), DialogPane#lookupButton does not return the node which is shown inside the button bar.
This is due DialogPane adding two list change listeners to 'buttons' (#getButtonTypes). They have the wrong order, which will result in the button bar not changing at all and the 'buttonNodes' list will recreate the dialog button(s).
Finally, this will make DialogPane#lookupButton returning the 'wrong' button, which is in fact not used inside the dialog button bar.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8263807](https://bugs.openjdk.java.net/browse/JDK-8263807): Button types of a DialogPane are set twice, returns a wrong button

### Reviewers
 * [Ajit Ghaisas](https://openjdk.java.net/census#aghaisas) (@aghaisas - **Reviewer**)
 * [Kevin Rushforth](https://openjdk.java.net/census#kcr) (@kevinrushforth - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jfx pull/432/head:pull/432` \
`$ git checkout pull/432`

Update a local copy of the PR: \
`$ git checkout pull/432` \
`$ git pull https://git.openjdk.java.net/jfx pull/432/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 432`

View PR using the GUI difftool: \
`$ git pr show -t 432`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jfx/pull/432.diff">https://git.openjdk.java.net/jfx/pull/432.diff</a>

</details>

### CODE REVIEW

**Review by aghaisas — APPROVED** [review]

**Review by kevinrushforth — COMMENTED** [review]
I have one question on the fix (see below). Also, have you run all of the unit tests (not just the new one)?

File: modules/javafx.controls/src/main/java/javafx/scene/control/DialogPane.java [hunk 1]
```diff
@@ -1057,7 +1059,7 @@ private void updateButtons(ButtonBar buttonBar) {
 
         boolean hasDefault = false;
         for (ButtonType cmd : getButtonTypes()) {
-            Node button = buttonNodes.computeIfAbsent(cmd, dialogButton -> createButton(cmd));
+            Node button = buttonNodes.get(cmd);
```
[the remark this answers is not in the document]
Comment (Maran23) [reply modules/javafx.controls/src/main/java/javafx/scene/control/DialogPane.java:1062]: This change is not needed for the fix. 

I removed it, because #computeIfAbsent() isn't needed anymore.
The (now) first listener will run and create/remove buttons based off the changes to #getButtonTypes().
The second listener (which is calling this method) will use all created buttons (to put them inside the ButtonBar).

-> So only a simple #get() is needed.
#computeIfAbsent() basically masked this bug before, as it was also creating a button, if missing. But as mentionend, when we are here, all buttons should be already created by the first listener.

**Comment by Maran23** [comment]
> 
> 
> I have one question on the fix (see below). Also, have you run all of the unit tests (not just the new one)?

I did run all the normal tests and ever dialog related system test. Everything passed.
Isn't jcheck running all the tests anyway?

**Comment by kevinrushforth** [comment]
Thanks for the explanation. I'll finish my review shortly.

>  Isn't jcheck running all the tests anyway?

You mean GitHub Actions (jcheck doesn't build or run any tests). It would, but for two things:

1. You used the `master` branch of your repo, which is not recommended, and will cause problems for you if you ever do another pull request. See [this comment](https://github.com/openjdk/jfx/pull/432#issuecomment-801982830) added by the Skara bot.
2. You haven't enabled GitHub action on your personal fork of the `jfx` repository. See [wiki.openjdk.java.net/display/SKARA/Testing](https://wiki.openjdk.java.net/display/SKARA/Testing) for information on how to enable it.

File: modules/javafx.controls/src/test/java/test/javafx/scene/control/DialogPaneTest.java [hunk 2]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2021, Oracle and/or its affiliates. All rights reserved.
+ * Copyright (c) 2021, 2021, Oracle and/or its affiliates. All rights reserved.
```
Comment (kevinrushforth) [inline modules/javafx.controls/src/test/java/test/javafx/scene/control/DialogPaneTest.java:2]: You need to revert this change. There must be only a single year listed if the start year and last modified year are the same.

**Review by kevinrushforth — COMMENTED** [review]
The fix and test look fine. There is one needed change as noted below.

**Comment by Maran23** [comment]
> 
> 
> Thanks for the explanation. I'll finish my review shortly.
> 
> > Isn't jcheck running all the tests anyway?
> 
> You mean GitHub Actions (jcheck doesn't build or run any tests). It would, but for two things:
> 
>     1. You used the `master` branch of your repo, which is not recommended, and will cause problems for you if you ever do another pull request. See [this comment](https://github.com/openjdk/jfx/pull/432#issuecomment-801982830) added by the Skara bot.
> 
>     2. You haven't enabled GitHub action on your personal fork of the `jfx` repository. See [wiki.openjdk.java.net/display/SKARA/Testing](https://wiki.openjdk.java.net/display/SKARA/Testing) for information on how to enable it.

Oh okay, thanks for the clarification. Interesting enough I don't need to enable Github actions, they are already enabled for all other branches I pushed. I think because I forked OpenJFX after you added them. 
But as you mentioned they are not running for master. I pushed a separate branch based of master (so only this 2 commits made by me), the Github actions were successful:
https://github.com/Maran23/jfx/actions/runs/723063232

**Review by kevinrushforth — APPROVED** [review]

**Comment by kevinrushforth** [comment]
> Interesting enough I don't need to enable Github actions, they are already enabled for all other branches I pushed. I think because I forked OpenJFX after you added them.

Yes, if you fork after GitHub actions were added, then no additional action should be needed.

**Review by aghaisas — APPROVED** [review]

**Review by aghaisas — APPROVED** [review]
